src/TorchSharp/Tensor/TensorExtensionMethods.cs

```diff
@@ -105,13 +105,22 @@ internal static bool IsFloatingPoint(this ScalarType type)
         /// <param name="writer"></param>
         public static void Save(this Tensor tensor, System.IO.BinaryWriter writer)
         {
+            bool copied = false;
+
+            if (tensor.device_type != DeviceType.CPU) {
+                tensor = tensor.to(torch.CPU);
```

Comment:
Looking at the docs [`tensor->to`](https://pytorch.org/cppdocs/api/classat_1_1_tensor.html#_CPPv4NK2at6Tensor2toEN3c108optionalIN2at10ScalarTypeEEEN3c108optionalIN2at6LayoutEEEN3c108optionalIN2at6DeviceEEEN3c108optionalIbEEbbN3c108optionalIN2at12MemoryFormatEEE) might might be moving tensor by default.

Can you add a test to verify GPU tensor remains on GPU after this call?

Reply:
BTW, might make sense to expose that `copy` parameter to C# too to make its default value clear.

Reply:
The copy parameter only affects the semantics when the device and type is already the same as the intended target, so in this case, it should not affect anything.

That said, it might be useful to expose the parameter in the .NET API for 'to'.

Reply:
> Can you add a test to verify GPU tensor remains on GPU after this call?

TestSaveLoadGruOnCUDA() should already be doing that.